docker/wrapper.py

```diff
@@ -12,37 +12,131 @@
+            print(e.message, file=sys.stderr)
+            exit(e.returncode)
+        else:
+            log.info('mesos-master process running in background')
+
+
+    unique_toil_dir = 'Toil-RNAseq-' + str(uuid4())
+    work_dir = os.path.join(mount, unique_toil_dir)
+
     os.makedirs(work_dir)
     log.info('Temporary directory created: {}'.format(work_dir))
     config_path = os.path.join(work_dir, 'toil-rnaseq.config')
-    job_store = os.path.join(args.resume, 'jobStore') if args.resume else os.path.join(work_dir, 'jobStore')
+
+    output_dir = mount
+    if args.auto_scale:
+        #set job store to be clould URL, e.g. S3 bucket
+        job_store = os.path.join(args.resume, 'jobStore') if args.resume else args.job_store
+        output_dir = args.output_location
+    else:
+        job_store = os.path.join(args.resume, 'jobStore') if args.resume else os.path.join(work_dir, 'jobStore')
+   
+
     with open(config_path, 'w') as f:
-        f.write(generate_config(args.star, args.rsem, args.kallisto, mount,
+        f.write(generate_config(args.star, args.rsem, args.kallisto, args.hera, output_dir,
                                 args.disable_cutadapt, args.save_bam, args.save_wiggle,
-                                args.bamqc))
+                                args.bamqc, args.max_sample_size))
     loglevel = log.getEffectiveLevel()
 
-    command = ['toil-rnaseq', 'run',
+    if args.auto_scale:
+        #Pick the working directory carefully; it must exist on the workers
+        #Do not let Toil pick the directory because this will not work when
+        #launching the workflow with Dockstore because dockstore uses cwltool
+        #which makes the file system read only which will prevent Toil from 
+        #creating the directory. We cannot use a directory like
+        #/home/ubunut/... becuase that does not exist on coreos which is what
+        #the worker nodes use for an OS; '/tmp' will usually already be 
+        #created on most Linux distributions
+        command = ['toil-rnaseq', 'run',
+               job_store,
+               '--config', config_path,
+               '--workDir', '/tmp',
+               args.toilLoggingOption,
+               '--retryCount', '1000000']
+    else:
+        command = ['toil-rnaseq', 'run',
                job_store,
                '--config', config_path,
                '--workDir', work_dir,
                args.toilLoggingOption,
                '--retryCount', '1']
+
+
     if args.resume:
         command.append('--restart')
     if args.cores:
         command.append('--maxCores={}'.format(args.cores))
-    path_to_manifest = generate_manifest(args.sample_tar, args.sample_single, args.sample_paired, work_dir, args.output_basename)
+    path_to_manifest = generate_manifest(args.sample_tar, args.sample_single, \
+        args.sample_paired, work_dir, args.output_basenames)
     command.append('--manifest=' + path_to_manifest)
+
+    if args.auto_scale:
```

Comment:
Fourth `if args.auto_scale` in the function — if it can be done in one block that'd be preferable.

Reply:
I'll have to tackle that in a future release